ssl/ssl_test.cc

```diff
@@ -11852,10 +11848,22 @@ TEST(SSLTest, SSLFileTests) {
   };
 
   using ScopedFILE = std::unique_ptr<FILE, fclose_deleter>;
+
+#if defined(OPENSSL_WINDOWS)
+  OPENSSL_BEGIN_ALLOW_DEPRECATED
+  // |tmpnam| is deprecated due to security concerns, but we only use this to
+  // run tests.
   char rsa_pem_filename[L_tmpnam];
   char ecdsa_pem_filename[L_tmpnam];
   ASSERT_TRUE(tmpnam(rsa_pem_filename));
   ASSERT_TRUE(tmpnam(ecdsa_pem_filename));
+  OPENSSL_END_ALLOW_DEPRECATED
+#else
+  char rsa_pem_filename[] = "/tmp/fileXXXXXX";
+  char ecdsa_pem_filename[] = "/tmp/fileXXXXXX";
+  ASSERT_TRUE(mkstemp(rsa_pem_filename));
+  ASSERT_TRUE(mkstemp(ecdsa_pem_filename));
+#endif
 
   ScopedFILE rsa_pem(fopen(rsa_pem_filename, "w"));
   ScopedFILE ecdsa_pem(fopen(ecdsa_pem_filename, "w"));
```

Comment:
If we don't care about the file name, can't you just use  `tmpfile()` [as in bio_test.cc](https://github.com/aws/aws-lc/blob/main/crypto/bio/bio_test.cc#L176)? It should be [portable enough](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/tmpfile?view=msvc-170). These lines would just become:

```
ScopedFile rsa_pem(tmpfile());
ScopedFile ecdsa_pem(tmpfile());
```

Reply:
That's the conundrum I've been trying to solve with these tests :( 
`SSL_CTX_use_certificate_chain_file` and `SSL_use_certificate_chain_file` directly take the file name as the second argument and parse the certificates for the user. `tmpfile` was the initial plan, but it doesn't provide an interface to extract the created file name.

Reply:
Ah, I see. That makes sense.